app/Http/Controllers/SupplyController.php

```diff
@@ -0,0 +1,269 @@
+
+        $response = new StreamedResponse();
+        $response->setCallback(function () use ($jsPath, $request, $params) {
+            echo str_replace([
+                "'{{ ORIGIN }}'",
+                "'{{ ADUSER }}'",
+            ], $params, file_get_contents($jsPath));
+        });
+
+        $response->headers->set('Content-Type', 'text/javascript');
+
+        $response->setCache(array(
+            'etag' => md5(md5_file($jsPath) . implode(':', $params)),
+            'last_modified' => new \DateTime('@' . filemtime($jsPath)),
+            'max_age' => 3600 * 24 * 30,
+            's_maxage' => 3600 * 24 * 30,
+            'private' => false,
+            'public' => true
+        ));
+
+        if (! $response->isNotModified($request)) {
+            // TODO: ask Jacek
+        }
+        return $response;
+    }
+
+    public function logNetworkClick(Request $request, $id)
+    {
+        if ($request->query->get('r')) {
+            $url = Utils::UrlSafeBase64Decode($request->query->get('r'));
+            $request->query->remove('r');
+        } else {
+            $banner = NetworkCampaign::getRepository($this->getDoctrine()->getManager())->findOneBy([
+                'uuid' => $id
+            ]);
+
+            if (! $banner) {
+                throw new NotFoundHttpException();
+            }
+
+            $url = $banner->getClickUrl();
+        }
+        $qString = http_build_query($request->query->all());
+        if ($qString) {
+            $qPos = strpos($url, '?');
+
+            if ($qPos === false) {
+                $url .= '?' . $qString;
+            } elseif ($qPos == strlen($url) - 1) {
+                $url .= $qString;
+            } else {
+                $url .= '&' . $qString;
+            }
+        }
+
+        $logIp = bin2hex(inet_pton($request->getClientIp()));
+
+
+        $cid = Utils::getRawTrackingId($request->query->get('cid'));
+        $tid = Utils::getRawTrackingId($request->cookies->get('tid')) ?: $logIp;
+        // TODO get user / website / zone
+        $payFrom = $request->query->get('pfr');
+
+        $log = new NetworkEventLog();
+        $log->setCid($cid);
+        $log->setBannerId($id);
+        $log->setPayFrom($payFrom);
+        $log->setTid($tid);
+        $log->setIp($logIp);
+        $log->setEventType("click");
+        //         $log->setContext(Utils::getImpressionContext($this->container, $request));
+
+        $em = $this->getDoctrine()->getManager();
+        $em->persist($log);
+        $em->flush();
+
+        $url = Utils::addUrlParameter($url, 'pid', $log->getId());
+
+        $adselectService = $this->container->has('adselect') ? $this->container->get('adselect') : null;
+        $adselectService instanceof Adselect;
+
+        if ($adselectService) {
+            $adselectService->addImpressions([
+                $log->getAdselectJson(),
+            ]);
+        }
+
+        $response = new RedirectResponse($url);
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Immediately return this expression instead of assigning it to the temporary variable "$response". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1488)